### PR TITLE
Add Unit Tests for Helper Functions in helpers.go

### PR DESCRIPTION
**Pull request by ganeshvenkatasai**
### Description

This PR adds comprehensive unit tests for the helper functions defined in the `helpers.go` file. The following changes have been made:

- **Added Unit Tests:**
  - `MergeStringMaps`: Tests various scenarios including merging multiple maps, handling empty maps, and overwriting values with duplicate keys.
  - `ResolveFSRef`: Verifies correct resolution of file paths and handling of errors for invalid paths.

- **Testing Approach:**
  - Used table driven tests to ensure clarity and coverage of edge cases.
  - Verified function outputs against expected results.
  - Ensured all tests pass with no failures or errors.

- **Impact:**
  - Improves test coverage and code quality by catching potential bugs in the helper functions.

- **Additional Notes:**
  - No changes were made to the core logic of the functions.
  - All tests pass with the current implementation.

Please review the PR and provide feedback. Let me know if any additional modifications are needed!

### CODE REVIEW

**Comment by Ashparshp** [comment]
Thank you @ganeshvenkatasai,
let's discuss this PR during the Meshery dev meeting call, if you would. Adding it as an agenda item to the meeting minutes.

Timing: 7:30 PM IST / 9:00 AM CT (Wednesday)
Meeting minutes: [Google Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit)
Meeting link: [meet.layer5.io/meshery](https://meet.layer5.io/meshery)

**Comment by leecalcote** [comment]
Thanks, @ganeshvenkatasai!

File: server/helpers/helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	for k, v := range expected {
+		if result[k] != v {
+			t.Errorf("Expected key %s to have value %s, but got %s", k, v, result[k])
+		}
+	}
+}
+
+func TestResolveFSRef(t *testing.T) {
```
Comment (lekaf974) [inline server/helpers/helpers_test.go:26]: even if this increase code coverage I am not sure the test it self is relevant here and should have already been covered in meshkit

**Comment by lekaf974** [comment]
Thanks @ganeshvenkatasai

**Review by lekaf974 — APPROVED** [review]

File: server/helpers/helpers_test.go (hunk 1 above)
Comment (MUzairS15) [reply server/helpers/helpers_test.go:26]: +1 @lekaf974

Comment (MUzairS15) [reply server/helpers/helpers_test.go:26]: @ganeshvenkatasai PLease move to meshkit and also add file ref for "HTTP"

File: server/helpers/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package helpers
+
+import (
+	"os"
+	"testing"
+)
+
+func TestMergeStringMaps(t *testing.T) {
```
Comment (MUzairS15) [inline server/helpers/helpers_test.go:8]: @ganeshvenkatasai I thought you also added tests for empty maps and merging maps (not only overriding).

**Review by MUzairS15 — COMMENTED** [review]
@ganeshvenkatasai This is good but instead of unit tests, it is better if you focus on Integration tests for the server. Also, you can drive these efforts for "mesheryctl"/

File: server/helpers/helpers_test.go (hunk 2 above)
Comment (lekaf974) [reply server/helpers/helpers_test.go:8]: +1

**Comment by lekaf974** [comment]
@ganeshvenkatasai any update on this ?
